wezterm-gui/src/termwindow/spawn.rs

```diff
@@ -127,7 +128,10 @@ impl super::TermWindow {
         } else {
             match cwd {
                 Some(url) if url.scheme() == "file" => {
-                    let path = url.path().to_string();
+                    let path = percent_decode_str(url.path())
+                        .decode_utf8()
+                        .unwrap()
```

Comment:
We can't tolerate an unwrap here; it will kill everything that the user is doing if a program has set a bogus cwd.

A failure to decode to utf8 should result in the `path` being treated as `None`.

There's a similar section of code over there that should also have the same treatment:
https://github.com/wez/wezterm/blob/main/wezterm-mux-server-impl/src/sessionhandler.rs#L615